asustor_main.c

```diff
@@ -401,25 +450,52 @@ static struct asustor_driver_data asustor_fs6706_driver_data = {
 };
 
 /*
- * It currently looks like the older systems are easier to tell apart, at least if one doesn't insist
- * on detecting the 2 vs 4 vs 6 drives versions (I only did this for AS67xx because I had to do the
- * advanced detection anyway)
+ * AS66xx and AS5304 (maybe all 53xx?) can't be told apart just by DMI either,
+ * (both are "Insyde" "GeminiLake"), so we need to match PCI IDs here as well
  */
+static struct asustor_driver_data asustor_5304_driver_data = {
+	.name = "AS53xx",
+	.pci_matches = {
+		// ASMedia ASM2806 4-Port PCIe x2 Gen3 Packet Switch
+		// (does NOT exist here, but on AS602T and AS604T)
+		{ 0x1b21, 0x2806, 0, 0 }
+	},
+	.leds = &asustor_5304_gpio_leds_lookup,
+	.keys = &asustor_6100_gpio_keys_lookup,
+};
 
 static struct asustor_driver_data asustor_6600_driver_data = {
+	.name = "AS66xx",
+	.pci_matches = {
+		// ASMedia ASM2806 4-Port PCIe x2 Gen3 Packet Switch
+		// (exists on AS602T and AS604T, but not AS5304T)
+		// in the lspci outputs I've seen both had 5 such devices,
+		// but let's be a bit more flexible here..
```

Comment:
Is this to cover known 6604T but unknown 6602T, or are both known? If it's always 5 on _both_, I'd err on strictness, a false positive could be worse (changing random GPIOs) than a false negative.

Reply:
Also, we generally don't worry about false positives - after all, most devices are only matched by DMI which has an even higher chance of false positives.
So unless an incompatible device turns up that is wrongly matched here (and can be told apart by the number of PCIe switches) I wouldn't worry about that